src/lightning_lite/utilities/types.py

```diff
@@ -77,3 +77,11 @@ def __init__(
 
     def step(self, metrics: Union[float, int, Tensor], epoch: Optional[int] = None) -> None:
         ...
+
+
+@runtime_checkable
+class Steppable(Protocol):
+    """To structurally type ``optimizer.step()``"""
+
+    def step(self, closure: Optional[Callable[[], float]] = ...) -> Optional[float]:
```

Comment:
what about the kwargs? we need to be able to pass them down

Reply:
could we do this with overloads so that still one of the "options" would match the optimizer?

Reply:
@carmocca ok, so we would need to put the type ignore in the usages where we pass kwargs i guess

Reply:
Feel free to resolve my part of this conversation or maybe try what Justus said.